src/main/java/com/autotune/analyzer/recommendations/model/GenericRecommendationModel.java

```diff
@@ -495,6 +495,11 @@ public static JSONObject calculateNamespaceMemoryUsage(IntervalResults intervalR
     @Override
     public Map<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> getAcceleratorRequestRecommendation(Map<Timestamp, IntervalResults> filteredResultsMap, ArrayList<RecommendationNotification> notifications) {
 
+        boolean setNotification = true;
+        if (null == notifications) {
```

Comment:
We initialize notifications list before calling this method using
`ArrayList<RecommendationNotification> notifications = new ArrayList<>();`

Since this guarantees that the object will never be null, do we still need the null check here ?

Reply:
Yes, but the function should not be tightly dependent on this assumption. Also it would be a good practice to check the params passed to the function. Please let me know your thoughts on this.